include/ht_refcounted.h

```diff
@@ -93,9 +94,11 @@ namespace Hatchit
             Handle(VarType* varPtr, uint32_t* refCounter, const Guid* name);
 
             //Private members
-            VarType* m_ptr;
-            uint32_t* m_refCount;
+            VarType*    m_ptr;
+            uint32_t*   m_refCount;
             const Guid* m_ID;
+
+            mutable std::mutex m_mutex;
```

Comment:
Yes, the handle should not have the mutex. the refcount could be an atomic, and the guid could be readonly anyway. Its really the refcounted object that needs to be threadsafe.